docs/source/cli.rst

```diff
@@ -66,32 +71,148 @@ CLI and Client Examples
 Following sections present different typical usage of the :ref:`cli_commands` and :ref:`client_commands`.
 Operations are equivalent between the :term:`CLI` and Python client.
 
-Note that more operations and option parameters are available,
-and are not all necessarily represented in below examples.
+.. note::
+    Much more operations and option parameters are available.
+    They are not all necessarily represented in below examples.
+    Explore available arguments using ``weaver <operation> --help`` or
+    using the above documentation for :ref:`client_commands` and `cli_commands`.
 
 For each of the following examples, the client is created as follows:
 
 .. code-block:: python
+    :caption: Python
 
     client = WeaverClient(url="{WEAVER_URL}")
 
 
+.. _cli_example_auth:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Authentication Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For any operation that requires authentication/authorization to access a protected service targeted by ``{WEAVER_URL}``,
+it is possible to either provide the ``auth`` parameter during initialization of the :py:class:`weaver.cli.WeaverClient`
+itself (the specific :py:class:`weaver.cli.AuthHandler` is reused for **all** operations), or as argument to individual
+methods when calling the respective operation (handler only used for that step).
+
+Any class implementation that derives from :py:class:`weaver.cli.AuthHandler` or :py:class:`requests.auth.AuthBase` can
+be used for the ``auth`` argument. This class must implement the ``__call__`` method taking a single ``request``argument
+and returning the adjusted ``request`` instance. The call should apply any relevant modifications to grant the necessary
+authentication/authorization details. This ``__call__`` will be performed inline prior to sending the actual request
+toward the service.
+
+.. note::
+    There are multiple predefined handlers available for use:
+
+    - :py:class:`requests.auth.HTTPBasicAuth`
+    - :py:class:`requests.auth.HTTPProxyAuth`
+    - :py:class:`requests.auth.HTTPDigestAuth`
+    - :py:class:`weaver.cli.BasicAuthHandler`
+    - :py:class:`weaver.cli.BearerAuthHandler`
+    - :py:class:`weaver.cli.CookieAuthHandler`
+    - |requests-magpie-auth|_
+
+.. |requests-magpie-auth| replace:: ``requests_magpie.MagpieAuth``
+.. _requests-magpie-auth: https://github.com/Ouranosinc/requests-magpie/blob/master/requests_magpie.py
+
+When using the :ref:`cli_commands`, the specific :py:class:`weaver.cli.AuthHandler` implementation to employ
+must be provided using the ``--auth-handler`` (``-aH``) argument. This can be an importable (installed) class module
```

Comment:
Shouldn't the correct argument be `-aC AUTH_HANDLER_CLASS`?

`-aH HEADER` is for `--auth-header`, unless the typos are in the CLI itself.

Reply:
You're right. Forgotten after `--auth-header` was added.

Reply:
@francisPLT  Was your review completed?

Reply:
Almost! Tested the new features on a DACCS deployment with the MagpieAuth handler, everything went well and behaved as expected, just finishing up on the docs and CLI code

Reply:
I tried capabilities, deploy, deploy( the same process, with the -D option), undeploy, execute, status and results. The auth part worked great in all of those. 

I didn't try the link related options though, but I didn't have problems with the links provided by the operations I tried.

I did have an SSL error, but I'm pretty sure it's related to the age of my instance's certificate (IaC launched test instance, switched off ssl for weaver for testing)

Reply:
The `execute` worked on DACCS? Was the process only executable by the user used for the auth? I'm pleasantly surprised if this is the case.

Reply:
With a minimal permission user (Group: Users), I received this error when I tried to only give write permissions for the process route : `"weaver.owsexceptions.OWSNoApplicableCode: Failed to retrieve WPS capabilities.`

I ended up having to give a `Recursive Read` permission at top level (`weaver/`) for it to work, on top of the `Recursive Write` for the specific process route (`weaver/processes/name-of-process`).

There's probably a way to configure it more tightly, but it's still a step more than running with with the admin account!